pvlib/modelchain.py

```diff
@@ -1200,7 +1423,9 @@ def prepare_inputs_from_poa(self, data):
         --------
         pvlib.modelchain.ModelChain.prepare_inputs
         """
-
+        self._check_length(data)
+        if isinstance(data, tuple):
+            _validate_weather_indices(data)
```

Comment:
This pattern pops up a few places. Should refactor (I think fold the index validation in to the length check).